bpo-29554: Improve docs for pstat module and profile.

Carreau opened the pull request:
Clarify that methods takes a string which is interpreted as a re, and
not a re object.

Clarify what the old `-1`, `0`, `1` and `2` option were.

bpo-29554

Carreau commented on Doc/library/profile.rst:
```diff
@@ -444,9 +444,10 @@ Analysis of the profiler data is done using the :class:`~pstats.Stats` class.
-      select a percentage of lines), or a regular expression (to pattern match
-      the standard name that is printed.  If several restrictions are provided,
-      then they are applied sequentially.  For example::
+      select a percentage of lines), or a string that will interpreted as a
```
As the rewrapping make a lot of line changes: 
> or a **string that will interpreted as a** regular expression

zware replied:
I'd recommend redoing the PR as a commit to make the change(s) you want and a commit to rewrap.  That way the change can be reviewed more easily.

Carreau replied:
Ohhhh ! Why did I never thought of that. Done.

Carreau commented on Lib/pstats.py:
```diff
@@ -48,11 +48,13 @@ class Stats:
-    minor key of 'the name of the function'.  Look at the two tables in
-    sort_stats() and get_sort_arg_defs(self) for more examples.
+    addition to the old -1, 0, 1, or 2 that are respectively  interpreted as
+    'stdname', 'calls', 'time', and 'cumulative').  It takes an arbitrary
```
add:
> **that are respectively interpreted as 'stdname', 'calls', 'time', and 'cumulative'**

Carreau commented on Lib/pstats.py:
```diff
@@ -48,11 +48,13 @@ class Stats:
+    number of quoted strings to select the sort order.  For example
+    sort_stats('time', 'name') sorts on the major key of 'internal function
+    time', and on the minor key of 'the name of the function'.  Look at the two
+    tables in sort_stats() , get_sort_arg_defs(self) and sort_arg_dict_default
```
`and` to `,` and add `and sort_arg_dict_default`

Carreau committed:
Improve docs for pstat module and profile.

Clarify that methods takes a string which is interpreted as a re, and
not a re object.

Clarify what the old `-1`, `0`, `1` and `2` option were.

Carreau committed:
Rewrap paragraph

Re wrap as a second commit, to make review easier.

berkerpeksag commented:
This requires a new tracker item on bugs.python.org. See step 5 at http://cpython-devguide.readthedocs.io/#quick-start for details.

Carreau commented:
> This requires a new tracker item on bugs.python.org. See step 5 at http://cpython-devguide.readthedocs.io/#quick-start for details.

Thank you, I thought that was minor enough. but done as [bpo-29554](http://bugs.python.org/issue29554)

berkerpeksag commented on Doc/library/profile.rst:
```diff
@@ -444,9 +444,10 @@ Analysis of the profiler data is done using the :class:`~pstats.Stats` class.
-      the standard name that is printed.  If several restrictions are provided,
-      then they are applied sequentially.  For example::
+      select a percentage of lines), or a string that will interpreted as a
+      regular expression (to pattern match the standard name that is printed.
```
Shouldn't "(to pattern match the standard name that is printed." be end with a closing parenthesis ?

Carreau replied:
Yes. Done.

berkerpeksag commented on Lib/pstats.py:
```diff
@@ -48,11 +48,12 @@ class Stats:
-    'name') sorts on the major key of 'internal function time', and on the
-    minor key of 'the name of the function'.  Look at the two tables in
-    sort_stats() and get_sort_arg_defs(self) for more examples.
+    addition to the old -1, 0, 1, or 2 that are respectively  interpreted as
```
There are multiple whitespaces between "respectively  interpreted".

Carreau replied:
Good eyes !

berkerpeksag commented on Lib/pstats.py:
```diff
@@ -48,11 +48,12 @@ class Stats:
-    sort_stats() and get_sort_arg_defs(self) for more examples.
+    addition to the old -1, 0, 1, or 2 that are respectively  interpreted as
+    'stdname', 'calls', 'time', and 'cumulative').  It takes an arbitrary number
+    of quoted strings to select the sort order.  For example sort_stats('time',
```
Instead of breaking `sort_stats('time', 'name')` call I'd move the "For example [...]" sentence to its own paragraph.

Carreau replied:
Done as well.

berkerpeksag commented:
Thanks!

Carreau commented:
Thanks, much appreciated.
